### PR TITLE
Fixes #35: Add <firebase-auth> support for custom token auth

### LINKED ISSUE #35: signInWithCustomToken and just puzzled about ploymerfire's direction.
Hi,

Will signInWithCustomToken be supported?  Also, will there be something similar to `<firebase-collection>`? 

I'm currently using the 'official' firebase elements, https://elements.polymer-project.org/elements/firebase-element and am wondering what is status and direction with polymerfire. Will it eventually replace the original firebase-element?

Thanks.

### PR DESCRIPTION
**Pull request by sgammon**
This changeset adds a simple function to support `signInWithCustomToken` in the `<firebase-auth>` component.

### CODE REVIEW

**Comment by sgammon** [comment]
I signed it!

**Comment by sgammon** [comment]
It is a corp CLA under the name _Momentum Ideas Co._

File: firebase-auth.html [hunk 1]
```diff
@@ -122,6 +122,18 @@
...
+         * Authenticates a Firebase client using a custom JSON Web Token.
+         *
+         * @return {Promise} Promise that handles success and failure.
+         */
+        signInWithCustomToken: function(token) {
+          if (!this.auth) {
+            return Promise.reject('No app configured for firebase-auth!');
+          }
+          return this._handleSignIn(this.auth.signInWithCustomToken(token));
+        },
```
Comment (sgammon) [inline firebase-auth.html]: woop,
there it is

**Comment by myuseringithub** [comment]
Thanks man ! This is a must feature. Wondering why it wasn't included from first time.

**Comment by unusec** [comment]
Just fund this after I spent some time to create this on my own, this should be merged asap.

**Comment by mbleigh** [comment]
Great! Thanks for the PR.

**Comment by myuseringithub** [comment]
Thanks for merging it. I cannot figure out how to use it.

I'm getting the following error: 
POST https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyCustomToken?key=KEY 400 ()
page.php:903 aut error
firebase.js:75 Uncaught Error: The custom token format is incorrect. Please check the documentation.

Please provide a simple example. This is what I'm currently using: 
`    <firebase-app
      name="app"
      api-key="APIKEY"
      auth-domain="XXX.firebaseapp.com"
      database-url="https://XXX.firebaseio.com"
    ></firebase-app>
    <firebase-auth
      id="auth"
      app-name="app"
      provider="custom"
      on-error="_handleAuthError"
    ></firebase-auth>
    <firebase-query
      id="query"
      app-name="app"
      path="/path/subpath"
      data="{{data}}"
    ></firebase-query>
`
...
`ready: function() {
      this.$.auth.signInWithCustomToken('KEY');
},`

**Comment by sgammon** [comment]
well you will probably need a better key than `'KEY'`... lol

but I digress, Firebase's custom token format has changed. so if you are generating tokens in the old 2.x format (we were), you will have to switch over to their new token generator library.

see here for code samples in Java and Node.JS:
https://firebase.google.com/docs/auth/server#create_a_custom_token

basically they changed the token format so that all the developer-provided claims are listed in a `claims` hash. the JWT is also now signed with your service account's private key, instead of the simple Firebase secret like before.

**Comment by myuseringithub** [comment]
Oh didn't know that. I was actually using the old tokens. Generating them with the old PHP SDK for the server. 
That would be the problem then. Thanks a lot !

**Pull request merged**